server/backend/testspg/crud/test_user_validator.py

```diff
@@ -0,0 +1,138 @@
+import os
+import logging
+import pytest
+from fastapi import HTTPException
+
+from api.app.crud.validator.user_validator import UserValidator
+from api.app.schemas import Requester, TargetUser
+from api.app.constants import ERROR_CODE_INVALID_REQUEST_PARAMETER, UserType
+from testspg.constants import TEST_IDIR_REQUESTER_DICT, USER_NAME_BCEID_LOAD_2_TEST, USER_GUID_BCEID_LOAD_2_TEST
+
+
+LOGGER = logging.getLogger(__name__)
+
+# when run test locally, we can use our own idir username and user guid
+TEST_IDIR_USER_GUID = os.environ.get("TEST_IDIR_USER_GUID")
+TEST_IDIR_USER_NAME = os.environ.get(
+    "TEST_IDIR_USER_NAME"
+) or TEST_IDIR_REQUESTER_DICT.get("user_name")
+
```

Comment:
I only added the test for user validator in backend, cause backend already setup with real IDIR GUID. I didn't add the test in admin management because of duplication, if we feel we need, I can add it there as well.

Reply:
Cool! thanks Catherine.
Since we already had `IdimProxyService` test which uses real TEST_IDIR_USER_GUID, this one is a validator which uses `IdimProxyService`, then ideally we don't need to use real ID anymore here with real connection to IDIM-Proxy.
A proper mocking the `IdimProxyService` returns on (`search_idir` and `search_business_bceid`) for several test scenarios would be better. We probably don't have this mocking example yet, but might not be difficult (just need to use python mock package and mock the `IdimProxyService` import I guess), only the mock data is a little hard to come up with.
Would you want to give it a try?

Reply:
I was thinking of that before, but because of the edge case for information mismatch I want to use the real one for testing. I could do the mocked one in the admin management.

Reply:
Sure, but I am not sure what difference it would make for mismtach cases.
I can later see if I can do this improvement, because I feel the real IDIR ideally should only be used in real tests on external connection.